### PR TITLE
support dump scheduler cache snapshot to json file

### LINKED ISSUE #3154: dump scheduler snapshot into local file in a format that is easy to parse
#### What would you like to be added:
Currently, users can send signal to volcano by executing **kill -SIGUSR2 PID** to make volcano dump snapshot info log. However, this information is mixed with normal log and do not have a format. This makes it hard to parse and use this cache information , especially in a large cluster.
#### Why is this needed:
1. kill -SIGUSR2 PID make volcano dump the snapshot into a local file like volcano.timestamp.snapshot
2. the data in the file **volcano.timestamp.snapshot** can be parsed by tools like jq, which makes operations like **group by** or **order by** possible

### PR DESCRIPTION
**Pull request by lowang-bh**
Fixes #3154

### CODE REVIEW

**Review by hwdef — COMMENTED** [review]
Most of them are fine.

I think it would be best if we could specify the output location of the file via the parameter

**Comment by lowang-bh** [comment]
Test results.
image

**Comment by lowang-bh** [comment]
> I think it would be best if we could specify the output location of the file via the parameter

Done。
image

**Review by hwdef — COMMENTED** [review]
/lgtm
Thanks

**Comment by lowang-bh** [comment]
re-trigger CI

**Review by hwdef — COMMENTED** [review]
/lgtm

File: cmd/scheduler/app/options/options.go [hunk 1]
```diff
@@ -133,6 +135,7 @@ func (s *ServerOption) AddFlags(fs *pflag.FlagSet) {
 	fs.BoolVar(&s.EnableMetrics, "enable-metrics", false, "Enable the metrics function; it is false by default")
 	fs.StringSliceVar(&s.NodeSelector, "node-selector", nil, "volcano only work with the labeled node, like: --node-selector=volcano.sh/role:train --node-selector=volcano.sh/role:serving")
 	fs.BoolVar(&s.EnableCacheDumper, "cache-dumper", true, "Enable the cache dumper, it's true by default")
+	fs.StringVar(&s.CacheDumpFileDir, "dump-dir", "/tmp", "The target dir where the json file put at when dump cache info to json file")
```
Comment (Monokaix) [inline cmd/scheduler/app/options/options.go:138]: cache-dump-dir is better.

Comment (lowang-bh) [reply cmd/scheduler/app/options/options.go]: DONE.

**Comment by Monokaix** [comment]
Hi, please squash and remove merged commit, and split the commit into fixing unmarshal err and support dump scheduler cache snapshot to json file.

**Review by Monokaix — COMMENTED** [review]
/lgtm

**Comment by Monokaix** [comment]
Hi, the issue mentioned about marsh error seems serious, so can you split it to a single pr so we can backport it to previous relase.

**Comment by lowang-bh** [comment]
image

**Comment by lowang-bh** [comment]
Any questions ? @Monokaix @william-wang

**Review by william-wang — APPROVED** [review]
/approve
